ddl/table.go

```diff
@@ -738,12 +739,39 @@ func onUpdateFlashReplicaStatus(t *meta.Meta, job *model.Job) (ver int64, _ erro
 	if err != nil {
 		return ver, errors.Trace(err)
 	}
-	if tblInfo.TiFlashReplica == nil || (tblInfo.TiFlashReplica.Available == available) {
+	if tblInfo.TiFlashReplica == nil || (tblInfo.TiFlashReplica.Available == available) ||
+		(tblInfo.ID != physicalID && available == tblInfo.TiFlashReplica.IsPartitionAvailable(physicalID)) {
 		return ver, nil
 	}
 
-	if tblInfo.TiFlashReplica != nil {
+	if tblInfo.ID == physicalID {
 		tblInfo.TiFlashReplica.Available = available
+	} else if pi := tblInfo.GetPartitionInfo(); pi != nil {
+		// Partition replica become available.
+		if available {
+			allAvailable := true
+			for _, p := range pi.Definitions {
+				if p.ID == physicalID {
+					tblInfo.TiFlashReplica.AvailablePartitionIDs = append(tblInfo.TiFlashReplica.AvailablePartitionIDs, physicalID)
+				}
+				allAvailable = allAvailable && tblInfo.TiFlashReplica.IsPartitionAvailable(p.ID)
+			}
+			tblInfo.TiFlashReplica.Available = allAvailable
+		} else {
+			// Partition replica become unavailable.
+			for i, id := range tblInfo.TiFlashReplica.AvailablePartitionIDs {
+				if id == physicalID {
+					newIDs := tblInfo.TiFlashReplica.AvailablePartitionIDs[:i]
+					newIDs = append(newIDs, tblInfo.TiFlashReplica.AvailablePartitionIDs[i+1:]...)
+					tblInfo.TiFlashReplica.AvailablePartitionIDs = newIDs
+					break
+				}
+			}
+			tblInfo.TiFlashReplica.Available = false
```

Comment:
How do we deal with it when the physical ID is not found in `tblInfo.TiFlashReplica.AvailablePartitionIDs`?

Reply:
Good catch, this line should put in line #765;

If physical ID is not found, do nothing, just ignore it will be ok.